blockchain/reactor.go

```diff
@@ -122,10 +125,55 @@ func (bcR *BlockchainReactor) AddPeer(peer p2p.Peer) {
 	}
 	// peer is added to the pool once we receive the first
 	// bcStatusResponseMessage from the peer and call pool.SetPeerHeight
+
+	// Getting the peer metric initializes it in the store
+	if bcR.Switch != nil {
+		if tms := bcR.Switch.MetricStore(); tms != nil {
```

Comment:
when it can be `nil`?

Reply:
the switch was nil while running tests, and the MetricStore doesn' t need to be set